src/subscription/checkout/SubscriptionCheckout.jsx

```diff
@@ -72,28 +70,26 @@ export const SubscriptionCheckout = () => {
     );
   };
 
-  useEffect(() => {
-    dispatch(fetchSubscriptionClientSecret());
-  }, [dispatch]);
-
   return (
     <section aria-label={intl.formatMessage(messages['subscription.checkout.payment.label'])}>
-      {shouldDisplayStripePaymentForm ? (
-        <>
-          <Elements options={options} stripe={stripePromise}>
-            <StripePaymentForm
-              options={options}
-              onSubmitPayment={handleSubmitStripe}
-              onSubmitButtonClick={handleSubmitStripeButtonClick}
-              isProcessing={stripeIsSubmitting}
-              isQuantityUpdating={isQuantityUpdating}
-              isSubscription
-              paymentDataSelector={subscriptionDetailsSelector}
-            />
-          </Elements>
-          <MonthlyBillingNotification />
-        </>
-      ) : ((loading || isClientSecretProcessing) && (<CheckoutSkeleton />))}
+      {
+        !loading ? (
+          <>
+            <Elements options={options} stripe={stripePromise}>
+              <StripePaymentForm
+                options={options}
+                onSubmitPayment={handleSubmitStripe}
+                onSubmitButtonClick={handleSubmitStripeButtonClick}
+                isProcessing={stripeIsSubmitting}
+                isQuantityUpdating={isQuantityUpdating}
```

Comment:
We don't need this for subscriptions, there is no quantity form that is updating (this is used in regular flow for bulk purchases)

Reply:
removed